Client/mods/deathmatch/logic/luadefs/CLuaCameraDefs.cpp

```diff
@@ -443,13 +445,18 @@ int CLuaCameraDefs::GetCameraClip(lua_State* luaVM)
 
 int CLuaCameraDefs::SetCameraViewMode(lua_State* luaVM)
 {
-    unsigned short   usViewMode = 0;
+    unsigned short   usVehicleViewMode = 0;
+    unsigned short   usPedViewMode = 0;
     CScriptArgReader argStream(luaVM);
-    argStream.ReadNumber(usViewMode);
+    argStream.ReadNumber(usVehicleViewMode);
+    argStream.ReadNumber(usPedViewMode, 0);
 
     if (!argStream.HasErrors())
     {
-        CStaticFunctionDefinitions::SetCameraViewMode(usViewMode);
+        CStaticFunctionDefinitions::SetCameraVehicleViewMode(usVehicleViewMode);
+
+        if (usPedViewMode != 0)
+            CStaticFunctionDefinitions::SetCameraPedViewMode(usPedViewMode);
 
         lua_pushboolean(luaVM, true);
         return 1;
```

Comment:
Please convert to use new argument parser. See `Shared/.../CLuaCryptoDefs.cpp` for an example.

Reply:
After all changes i have code like that
```
bool CLuaCameraDefs::SetCameraViewMode(eVehicleCamMode usVehicleViewMode, ePedCamMode usPedViewMode)
{
    CStaticFunctionDefinitions::SetCameraVehicleViewMode(usVehicleViewMode);

    if (usPedViewMode != 0) // here is wrong,  i know
        CStaticFunctionDefinitions::SetCameraPedViewMode(usPedViewMode);

    return true;
}
```
How to make second argument optional here?

Reply:
std::optional

Reply:
This way works, but there is another issue. I can put any value in second argument and function will return true, also i can put any string in first argument and function will return true. I need to use _int_ instead _unsigned short_ otherwise the parser does not work. New parser will broke old behavior.
```
bool CLuaCameraDefs::SetCameraViewMode(int usVehicleViewMode, std::optional<int> usPedViewMode)
{
    CStaticFunctionDefinitions::SetCameraVehicleViewMode((eVehicleCamMode)usVehicleViewMode);

    if (usPedViewMode)
        CStaticFunctionDefinitions::SetCameraPedViewMode((ePedCamMode)usPedViewMode.value());
 
    return true;
}
```